src/core/interfaces/IRequestMethods.ts

```diff
@@ -0,0 +1,125 @@
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  DELETE(input: unknown): unknown;
+
+  /**
+   * The GET HTTP method which is called if the request to the resource is a
+   * GET request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  GET(input: unknown): unknown;
+
+  /**
+   * The OPTIONS HTTP method which is called if the request to the resource is a
+   * OPTIONS request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  HEAD(input: unknown): unknown;
+
+  /**
+   * The OPTIONS HTTP method which is called if the request to the resource is a
+   * OPTIONS request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  OPTIONS(input: unknown): unknown;
+
+  /**
+   * The PATCH HTTP method which is called if the request to the resource is a
+   * PATCH request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  PATCH(input: unknown): unknown;
+
+  /**
+   * The POST HTTP method which is called if the request to the resource is a
+   * POST request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  POST(input: unknown): unknown;
+
+  /**
+   * The PUT HTTP method which is called if the request to the resource is a
+   * PUT request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  PUT(input: unknown): unknown;
+
+  /**
+   * The TRACE HTTP method which is called if the request to the resource is a
+   * TRACE request. See the following for more information on this method.
+   *
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/TRACE
+   *
+   * @param input The input to help produce an output.
+   * @returns An output based on the input.
+   */
+  TRACE(input: unknown): unknown;
```

Comment:
im not sure either, but maybe let's try it and see how it feels and document pros and cons between the two?

honestly, while working with this and adding typings, i started to think adding typings here and in other places like the handlers were useless and that they should only be provided closer to user land. like i can't really think of a reason why typings would need to travel down into the internals if they're really only being used far away from the internals. idk what to think about this really, but i'll try out the generics you wrote

Reply:
The rationale for me saying this is that it is automatically infered if you pass a generic, otherwise the default `unknown` is used. They way it's currently, unknown is always used and you need to cast to a specific type.

Reply:
gonna have to punt on this because it's requiring more work than i thought